skactiveml/classifier/_wrapper.py

```diff
@@ -326,7 +327,31 @@ def _fit(self, fit_function, X, y, sample_weight=None, **fit_kwargs):
         return self
 
     def __sklearn_is_fitted__(self):
-        return hasattr(self, "is_fitted_")
+        if hasattr(self, "is_fitted_"):
+            return True
+
+        try:
+            check_is_fitted(self.estimator)
+        except NotFittedError:
+            return False
+
+        # set attributes that would be set by the fit function
+        self.is_fitted_ = True
+        self.estimator_ = deepcopy(self.estimator)
+        self.check_X_dict_ = {
+            "ensure_min_samples": 0,
+            "ensure_min_features": 0,
+            "allow_nd": True,
+            "dtype": None,
+        }
+
+        # initialize all attributes using dummy data
+        _ = self._validate_data(
+            X=np.zeros([1, self.estimator_.n_features_in_]),
+            y=[self.missing_label],
+            check_X_dict=self.check_X_dict_,
+        )
+        return True
```

Comment:
Introduce `is_pretrained` flag to handle potential issues with `check_n_features`.

Reply:
The _validate_data was not needed for SklearnClassifier and SklearnRegressor and could be removed without problems.